spec/spec_helper.rb

```diff
@@ -89,8 +89,28 @@
   # Changing this to `config.after(:each)` would
   # put this code inside the test scope, interfering
   # with the test execution.
+  #
+  # rubocop:disable Style/GlobalVars
   config.around do |example|
     example.run.tap do
+      # Stop reporting on background thread leaks after too many
+      # successive failures. The output is very verbose and, at that point,
+      # it's better to work on fixing the very first occurrences.
+      $background_thread_leak_reports ||= 0
+      if $background_thread_leak_reports >= 10
+        unless $background_thread_leak_warned ||= false
+          warn RSpec::Core::Formatters::ConsoleCodes.wrap(
+            "Too many leaky thread reports! Suppressing further reports.\n" \
+            'Consider addressing the previously reported leaks before proceeding.',
+            :red
+          )
+
+          $background_thread_leak_warned = true
+        end
```

Comment:
It may be worth adding a note at the end of the test suite execution too, otherwise if you ignore the output of running a lot of tests (which to be honest, is what I do with such a noisy output as we have right now), you may miss that we still have a bunch of those to fix :)